ADR/0022-secret-mgmt-for-user-workloads.md

```diff
@@ -26,65 +29,156 @@ Accepted
+
+#### Example: If RemoteSecret has to be created and uploaded without setting any target
+
 ```yaml
 apiVersion: appstudio.redhat.com/v1beta1
 kind: RemoteSecret
 metadata:
-  name: jdbc-connection-parameter-prod
+    name: test-remote-secret-secret
 spec:
-  secret:
-    environment: prod
-    name: jdbc-connection-parameter
- ...
+    target:
+status:
+  TBD
 ```
 
-#### Example: If the destination is `Namespace` and secret has to be linked to SA
+#### Example:If multiple working Secrets should be created along with RemoteSecret.
 ```yaml
-apiVersion: v1
-kind: Secret
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
 metadata:
-  name: jdbc-connection-parameter
-  annotations:
-      spi.appstudio.redhat.com/secret-linkedto-serviceAccount-name: sa-star
-  labels:
-    spi.appstudio.redhat.com/upload-secret: secret
-    spi.appstudio.redhat.com/target-namespace: ns-9459039345
-type: Opaque
-data:
- ...
+    name: test-remote-secret-secret
+spec:
+    secret:
+        name: test-remote-secret-secret
+    target:
+        - namespace: jdoe-test-ns
+        - namespace: jdoe-prod-ns
+status:
+    TBD
+```
+
+
+#### Example: If RemoteSecret has to be created with multiple target namespaces including BYC use-case
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: test-remote-secret-secret
+spec:
+    secret:
+        name: test-remote-secret-secret
+    target:
+        - namespace: jdoe-test-ns
+        - namespace: jdoe-prod-ns
+          apiURL: https://somedomain.copm:443
+          clusterCredentialsSecret: team-a--prod-dtc--secret
+status:
+    TBD
 ```
+
+#### Example: If RemoteSecret has to be created with target namespace and Environment
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: test-remote-secret-secret
+    labels:
+        appstudio.redhat.com/environment: prod
+        appstudio.redhat.com/component: m-service
+        appstudio.redhat.com/application: coffee-shop
+spec:
+    secret:
+        name: test-remote-secret-secret
+    target:
+        - namespace: jdoe-tenant
+status:
+    TBD
+```
+
+#### Example: If RemoteSecret has to be created all Environments of certain component and application
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: test-remote-secret-secret
+    labels:
+        appstudio.redhat.com/component: m-service
+        appstudio.redhat.com/application: coffee-shop
+spec:
+    secret:
+        name: test-remote-secret-secret
+    target:
+        - namespace: jdoe-tenant
+status:
+    TBD
+```
```

Comment:
Do we really need so many examples of creating RS with targets?
It makes reader think that is it the main usecase while the documentation tells opposite.
Does it makes sense to demonstrate more how to make RS with the Secret of different types, with or w/o SA instead?